testutil/containschecker_test.go

```diff
@@ -221,3 +221,72 @@ func (*containsCheckerSuite) TestDeepContainsUncomparableType(c *check.C) {
 	testCheck(c, DeepContains, true, "", containerSlice, elem)
 	testCheck(c, DeepContains, true, "", containerMap, elem)
 }
+
+type dummy struct {
+	a string
+	b map[string]int
+}
+
+func (*containsCheckerSuite) TestDeepContainerMatchesSliceSuccess(c *check.C) {
+	slice1 := []dummy{
+		{a: "one", b: map[string]int{"a": 1}},
+		{b: map[string]int{"b": 2}, a: "two"},
```

Comment:
Isn't the order of the initializers irrelevant in any case? I think that swapping the order makes this test harder to read and risks letting the reader miss the point of the test. I'd always initialize first `a` then `b`, and just swap the order of the elements between `slice1` and `slice2`.